### ISSUE TITLE
gnirehtet and android 9.1?

### ISSUE THREAD

**Issue by choong-sys** [Genymobile/gnirehtet#225]
Does gnirehtet work with android 9.1? I can't get mine working with it.

**Comment by rom1v** [comment]
Could you test current `master` branch?

https://github.com/Genymobile/gnirehtet/commit/cfc63a2c03bb10232ce04a687470d7594b4895f8

**Comment by choong-sys** [comment]
Tested current master branch, but not working as well, Could you please provide a solution?

**Comment by rom1v** [comment]
Note that you must uninstall first before testing master:

    ./gnirehtet uninstall

What do you mean by "not working"? Do you have the confirmation popup? Could you provide some relevant logs from `adb logcat`?

**Comment by choong-sys** [comment]
It doesn't show the confirmation popup, and I've attached the whole relevant, hope it can help. 
[file-2.txt](https://github.com/Genymobile/gnirehtet/files/3634934/file-2.txt)

**Comment by choong-sys** [comment]
What I did was, download the gnirehtet-rust-win64-v2.3 (Windows) & platform-tools_r29.0.2-windows folder from master version, combine them and run it with cmd. It works perfectly for android os below 9 but not for os 9. Please let me know if you need more information about this. 
You did mention that don't use master version, which version should I use to run it again?

**Comment by rom1v** [comment]
> What I did was, download the gnirehtet-rust-win64-v2.3 (Windows) & platform-tools_r29.0.2-windows folder from master version

Oh, that was the latest release then (referenced from the README.md, in the master branch).

> You did mention that don't use master version, which version should I use to run it again?

You need to build it, it's not released yet (there are other things to do before a new release): https://github.com/Genymobile/gnirehtet/blob/master/DEVELOP.md

**Comment by choong-sys** [comment]
Yup I was testing with your latest release as you mentioned to test your current master branch. 
Can i know which version of branch that i can use to build and run your latest code with Android Studio? And would you mind to tell me when will be your next release?

**Comment by rom1v** [comment]
> Yup I was testing with your latest release as you mentioned to test your current master branch.

The latest release is on tag `v2.3` (3104563fb915454565ac9c5cf613d7aca5ca5cb0). Of course, the link to the release in the README is unchanged until the next release.

> Can i know which version of branch that i can use to build and run your latest code with Android Studio?

`master` (currently 2a22b30d90b78cb4ee738c48f54ee330aa929f1b).

> And would you mind to tell me when will be your next release?

I don't know. When I have time to fix some other stuff, fix the cross-compilation for Windows, and release…

**Comment by choong-sys** [comment]
> `master` (currently [2a22b30](https://github.com/Genymobile/gnirehtet/commit/2a22b30d90b78cb4ee738c48f54ee330aa929f1b)).
> I have tried to download the file of this branch and combine with the platform-tools_r29.0.2-windows folder to run it, but it's still the same as previous testing. Is this commit version has included the fix for the problem of Android OS 9? 
FYI, this is the log I've tracked, 
[file_New.txt](https://github.com/Genymobile/gnirehtet/files/3635362/file_New.txt)

> I don't know. When I have time to fix some other stuff, fix the cross-compilation for Windows, and release…
> Alright, thanks for letting me know and good luck for your fixes. "Crossed Fingers"

**Comment by rom1v** [comment]
> I have tried to download the file of this branch

There is no file to "download", you need to `git clone` then build the project yourself to build `master`.

**Comment by choong-sys** [comment]
I'm trying to run it with Android Studio, buy it shows "Default Activity not found", anything that I can do to fix this?

**Comment by choong-sys** [comment]
I managed to add these two lines of code into AndroidManifest file 
```
<action android:name="android.intent.action.MAIN" />
<category android:name="android.intent.category.LAUNCHER" />
```
and installed the app successful into my physical phone but unable to launch it. Could you help on this ?

**Comment by rom1v** [comment]
You cannoy "run" gnirehtet as an Android app. Instead, just build the apk (`./gradlew build`) and the desktop (compand-line) app, then run the app.

**Comment by choong-sys** [comment]
I've tried to build the apk and run it with desktop (command-line) and run the app, but still not working in Android OS 9, works fine for OS lower than 9

**Comment by choong-sys** [comment]
Sir, I've tried build the apk and run the app with command-line app in Android OS 9, but it stuck at this step. Could you please help to fix this ? 

![image](https://user-images.githubusercontent.com/55570807/65931420-23b8e000-e43c-11e9-9fae-97e5765faeee.png)

**Comment by rom1v** [comment]
@choong-sys You need the current `master`, which fixes this issue. See #227 (https://github.com/Genymobile/gnirehtet/issues/227#issuecomment-535362883).

**Comment by choong-sys** [comment]
Yup, I've already pulled the latest changes of your current master branch in Android Studio and build the apk, then run it in Android OS 9, and it still showing the same result as `Broadcast completed: result=0`

**Comment by rom1v** [comment]
Ah, you need to `./gnirehtet uninstall` first (since this is not a release, the version has not been increased, so it does not reinstall the new version on the device).

If you see `Broadcast completed`, then this is not the latest version.

**Comment by rom1v** [comment]
And you must run with `./gnirehtet run` or `./gnirehtet start` (not `adb shell am broadcast …`).

**Comment by choong-sys** [comment]
I've uninstalled it and reinstall again before i run the apk with adb platform-tools, but it still showing the same result. Would you mind to send me the apk file that has fix this issue?  
![image](https://user-images.githubusercontent.com/55570807/65940929-5c68b180-e45c-11e9-86a8-28578827c199.png)

**Comment by rom1v** [comment]
In fact, the problem is the client executed on your computer. It's not the last version. How did you compile it? Is it the one accessible from your PATH?

**Comment by choong-sys** [comment]
I've tried 2 steps to compile and run it: 

1. Clone your git repo -> checked out from your latest master branch -> then build the apk -> place it into the adb platform-tools folder (downloaded from your git README.md) -> run the directory with cmd -> `./gnirehtet uninstall` -> `./gnirehtet run` -> it shows the same result 

2. Download the zip file from [https://github.com/Genymobile/gnirehtet/issues/227#issuecomment-535362883](url) -> extract the files -> get the apk file -> place that apk file into the adb platform-tools folder (downloaded from your git README.md) -> run the directory with cmd -> `./gnirehtet uninstall` -> `./gnirehtet run` -> it shows the same result

**Comment by rom1v** [comment]
OK, so you did not build the app executed on the computer (`gnirehtet.exe`).

**Comment by choong-sys** [comment]
How do i build the app to execute on the computer? 
Fyi, the 2 steps I've mentioned are working for Android OS < 9.1, but not working on OS >= 9.1

**Comment by choong-sys** [comment]
@rom1v this is revision that I've used to build the apk file and run it. It works fine for OS < 9.1 but not for OS >= 9.1 

![image](https://user-images.githubusercontent.com/55570807/66036224-4aac0a80-e53f-11e9-8c67-e8a4b059ae72.png)

**Comment by rom1v** [comment]
OK for the apk, but you need to build the relay server (`gnirehtet.exe`) from that version too.

https://github.com/Genymobile/gnirehtet/blob/master/DEVELOP.md#build

**Comment by choong-sys** [comment]
I have tried to build it, but build failed. This is the result. Can you help look into this please? 

![image](https://user-images.githubusercontent.com/55570807/66098165-2c441e80-e5d4-11e9-9ded-36c347e30ea9.png)
